packages/abi-coder/src/utils/json-abi.ts

```diff
@@ -3,6 +3,39 @@ import { ErrorCode, FuelError } from '@fuel-ts/errors';
 import type { ResolvedAbiType } from '../ResolvedAbiType';
 import type { JsonAbi, JsonAbiArgument, JsonAbiFunction, JsonAbiType } from '../types/JsonAbi';
 
+import {
+  DEFAULT_ENCODING_VERSION,
+  ENCODING_V0,
+  ENCODING_V1,
+  type EncodingVersion,
+} from './constants';
+
+/**
+ * Asserts that the encoding version is supported by the ABI coder.
+ *
+ * @param encoding - the encoding version to check
+ * @returns the encoding version
+ * @throws FuelError if the encoding version is not supported
+ */
+export const getEncodingVersion = (encoding?: string): EncodingVersion => {
+  switch (encoding) {
+    case undefined:
+      return DEFAULT_ENCODING_VERSION;
+
+    case ENCODING_V1:
+      return ENCODING_V1;
+
+    case ENCODING_V0:
+      return ENCODING_V0;
+
+    default:
+      throw new FuelError(
+        ErrorCode.UNSUPPORTED_ENCODING_VERSION,
+        `Encoding version '${encoding}' is unsupported.`
+      );
+  }
+};
```

Comment:
```suggestion
export const getEncodingVersion = (encoding?: string): EncodingVersion => {
  switch (encoding) {
    case undefined:
    case ENCODING_V0:
        return ENCODING_V0;

    case ENCODING_V1:
      return ENCODING_V1;

    default:
      throw new FuelError(
        ErrorCode.UNSUPPORTED_ENCODING_VERSION,
        `Encoding version '${encoding}' is unsupported.`
      );
  }
};
```
And you can remove the DEFAULT_ENCODING_VERSION constant with this.